tests/ValueFormatters/ValueFormatterTestBase.php

```diff
@@ -56,7 +56,7 @@ public function testValidFormat(
 			$formatter = $this->getInstance( $options );
 		}
 
-		$this->assertEquals( $expected, $formatter->format( $value ) );
+		$this->assertSame( $expected, $formatter->format( $value ) );
```

Comment:
This is the whole point of this patch. A test that expects the string `"9001.00"` but gets `"9001"` from the formatter currently does not fail, but must fail.

I can think of a formatter returning ints and floats, for example. But objects? Currently we have a single formatter that may return null. All others are guaranteed to return strings.

Put simple: If there is ever a formatter returning objects I believe it should *not* use this `ValueFormatterTestBase` base class then.